src/manifolds/ProbabilitySimplex.jl

```diff
@@ -421,16 +452,119 @@ function riemannian_gradient!(M::ProbabilitySimplex, X, p, Y; kwargs...)
 end
 
 @doc raw"""
     zero_vector(M::ProbabilitySimplex, p)
 
-returns the zero tangent vector in the tangent space of the point `p`  from the
+Return the zero tangent vector in the tangent space of the point `p`  from the
 [`ProbabilitySimplex`](@ref) `M`, i.e. its representation by the zero vector in the embedding.
 """
 zero_vector(::ProbabilitySimplex, ::Any)
 
 zero_vector!(::ProbabilitySimplex, X, p) = fill!(X, 0)
+
+@doc raw"""
+    RealProbabilityAmplitudes
+
+An explicit isometric embedding of interior of [`ProbabilitySimplex`] in
+positive orthant of the [`Sphere`]. Some properties extend to the boundary but not all.
+
+This embedding isometrically maps the Fisher-Rao metric on the open probability simplex to
+the sphere of radius 1 with Euclidean metric. More details can be found in Section 2.2
+of [^AyJostLeSchwachhöfer2017].
+
+The name derives from the notion of probability amplitudes in quantum mechanics.
+They are complex-valued and their squared norm corresponds to probability. This construction
+restricted to real valued amplitudes results in this embedding.
+
+# Constructor
+
+    RealProbabilityAmplitudes(n)
+"""
+const RealProbabilityAmplitudes{N} = EmbeddedManifold{ℝ,<:ProbabilitySimplex{N},Sphere{N,ℝ}}
+
+function RealProbabilityAmplitudes(n)
+    return EmbeddedManifold(ProbabilitySimplex(n), Sphere(n))
+end
+
+@doc raw"""
+    embed(M::RealProbabilityAmplitudes{N}, p) where {N}
+
+Convert point `p` on `ProbabilitySimplex` to (real) probability amplitude. The formula reads
+``(\sqrt{p_1}, \sqrt{p_2}, …, \sqrt{p_{N+1}})``. This is an isometry from the interior of
+the probability simplex to the interior of the positive orthant of a sphere.
+"""
+function embed(M::RealProbabilityAmplitudes, p)
+    return embed!(M, similar(p), p)
+end
+
+function embed!(::RealProbabilityAmplitudes, q, p)
+    q .= sqrt.(p)
+    return q
+end
+
+@doc raw"""
+    project(M::RealProbabilityAmplitudes{N}, p) where {N}
+
+Convert point (real) probability amplitude `p` on to a point on [`ProbabilitySimplex`](@ref).
+The formula reads ``(p_1^2, p_2^2, …, p_{N+1}^2)``. This is an isometry from the interior of
+the positive orthant of a sphere to interior of the probability simplex.
+"""
+function project(M::RealProbabilityAmplitudes, p)
+    return project!(M, similar(p), p)
+end
+
+function project!(::RealProbabilityAmplitudes, q, p)
+    q .= p .^ 2
+    return q
+end
+
+@doc raw"""
+    embed(M::RealProbabilityAmplitudes, p, X)
+
+Compute differential of [`embed`](@ref) on [`RealProbabilityAmplitudes`](@ref) at tangent
+vector `X` from the tangent space at `p` from sphere.
+"""
+function embed(M::RealProbabilityAmplitudes, p, X)
+    return embed!(M, similar(X), p, X)
+end
+
+function embed!(::RealProbabilityAmplitudes, Y, p, X)
+    Y .= X ./ sqrt.(p)
+    return Y
+end
+
+@doc raw"""
+    project(M::RealProbabilityAmplitudes, p, X)
+
+Compute differential of [`project`](@ref) on [`RealProbabilityAmplitudes`](@ref) at tangent
```

Comment:
This line might require an update?

Reply:
Or to be more precise – stating that a function that is called project – computes a differential of project (itself?) might be misleading.

Reply:
In this case I think understanding it as a differential of point projection is more appropriate, and understanding it as a projection from embedding is misleading.

Reply:
Hm.
* Then the double use of the name `project` is misleading
* A function that is not projection but called project is also a bit misleading
* I have no clue what this function does then ;)

Reply:
Maybe it should indeed be called `diff_project(M, p, X)` then?

Reply:
It technically is a projection, for a sufficiently broad definition of projection. And AFAIK some vector projections that we have are also differentials of point projections (f.e. sphere). This is a bit weird case because we embed in a space of the same dimension and point projection doesn't return the point closest in the geodesic sense. But otherwise it's technically fine.

Reply:
Je suis veldig verwirrt now (even in 4 languages).
The vector projection on the sphere actually agrees with projecting the vector onto the tangent space (interpreted as the plane parallel to the tangent plane running through zero of course).

So for now I am a bit lost what this function actually is. The docs say it is the differential of the point-projection (so I would call it `diff_project`).
So for me the most generic definition of a projection (say a function f) is that `f(f(p))=f(p)` or that `||f(p)-p|| is minimal compared to all other values f can attain (here in the tangent space).

Technically – I do not want to discuss and trust you – in confusion level … I would like to de-confuse both me and readers of these docs ;)

Reply:
I'm definitely interested in clarifying the confusion but I hope to do it correctly :slightly_smiling_face: .

> So for me the most generic definition of a projection (say a function f) is that `f(f(p))=f(p)` or that `||f(p)-p|| is minimal compared to all other values f can attain (here in the tangent space).

It's not the right definition here because because our `f` here has different domain and codomain. So `f(f(p))` is not defined, and you can't compare `p` and `f(p)`. This is already true for our embedding of SE(n) in GL(n+1), it works the same there. But it can indeed be confusing. Maybe using the names `project` and `embed` really isn't right.

Reply:
For embed (±change of representation in code/variable sense) you convinced me. Here you did not yet, though you are right, since we do have a change of the space (domain-range) my two notions were not that correct here. They are maybe more like (embed=id into the embedding) `f(embed(f(p)) == f(p)` and `embed(f(p))` is the closest point to `p` that embed can reach

Reply:
OK, then I think this `project(M, p)` is not a valid projection. It's just a local isometry and we need a different API :slightly_smiling_face: .

Reply:
I think in the very beginning I already said: What about `convert`? Because for me this looks like a conversion between two representations of the manifold?

Reply:
...hm. though that API is maybe also not yet developed to convert between manifolds of same dimension or isomorphically map.

Reply:
(sorry for so much discussion, just want to make sure I understand it myself as well)

Reply:
`convert` feels weird here because this is actually about establishing a local isometry, not providing a full new representation of the manifold.

Reply:
Well, in optical terms this is just “bump”, since we deform the plane to the spherical orthant (or back with project) and similarly adapt the metric for tangent vectors (that follow the same bump). But, hm, I am also missing a good name. `bump()` is not a good name ;)

Reply:
Hm, I'm not sure. If we are doing new interface anyway, then I'd prefer either function name or some argument to be specific to this particular local isometry. Something like `bump_local_isometry` or `local_isometry(bump(M, N), p)`

Reply:
Bump was far from a serious recommendation. I do not have a good idea for the name of the local isometry here.

Reply:
But why do you want the second variant (and what does bump represent there? An IsometryType?) why then not switch that to domain, range instead of somefunction(domain, range)?

Reply:
`bump` was an example there, a placeholder for the right word. What I'd like there is a way to handle multiple (local) isometries between two manifolds. For example, what if someone wanted an isometry between ProbabilitySimplex and negative orthant of the sphere?

Reply:
uff, I see. I was already out of creativity when I came up with `local_isometry` as a name, but you could use `AbstractIsometry` as a super type for the first argument and call this one `DefaultIsometry`? So we would do something similar to Embeddings

Reply:
So, something like `local_isometry(DefaultLocalIsometry(ProbabilitySimplex(N), Sphere(N)), p)`? I guess `local_isometry(ProbabilitySimplex(N), Sphere(N), p)` could be a shorter notation for the first one.

Reply:
That sounds reasonable and mixes both our ideas :)

Reply:
I've tried working out the `local_isometry` idea but it seems vastly over-complicated for what is needed here, and forces me to consider things that are irrelevant to this PR (how to do `local_isometry` generic enough). Can we just go back to simple functions specialized to this one manifold embedding? Something like `from_probability_amplitude` and `to_probability_amplitude` from the original commit but just with more appropriate names.

Reply:
I would prefer both domain and range in the name instead of once the domain once the range? `simplex_to_amplitude` and `amplitude_to_simplex` might not be the best names but indicate what I would prefer.

Reply:
Sure, those names sound fine :+1: .

Reply:
Then let's take those (and refine them later when we have a general scheme). And let's also keep them non-exported for now then.

Reply:
OK, done :slightly_smiling_face: